test/server.cpp

```diff
@@ -2008,3 +2014,273 @@ R"(const viewerSettings = {
+  "            <option value=\"en\">en</option>\n" \
+  "            <option value=\"eng\"" SELECTED_ENG ">English</option>\n" \
+  "          </select>\n" \
+  "        </div>\n" \
+  "        <div class=\"kiwixNav__select\">\n" \
+  "          <select name=\"category\" id=\"categoryFilter\" class='kiwixNav__kiwixFilter filter' form=\"kiwixSearchForm\">\n" \
+  "            <option value=\"\">All categories</option>\n" \
+  "            <option value=\"wikipedia\">Wikipedia</option>\n" \
+  "          </select>\n" \
+  "        </div>\n" \
+  "      </div>\n" \
+  "      <form id='kiwixSearchForm' class='kiwixNav__SearchForm' action=\"/ROOT%23%3F/nojs\">\n" \
+  "        <input type=\"text\" name=\"q\" placeholder=\"Search\" id=\"searchFilter\" class='kiwixSearch filter' value=\"\">\n" \
+  "        <input type=\"submit\" class=\"kiwixButton kiwixButtonHover\" value=\"Search\"/>\n" \
+  "      </form>\n" \
+  "    </div>\n"
+
+#define HOME_BODY_TEXT(X) \
+  "    <div class=\"kiwixHomeBody\">\n" \
+  "        \n" \
+  "        <div class=\"book__list\">\n" \
+  "        <h3 class=\"kiwixHomeBody__results\">" X " book(s)</h3>\n"
+
+#define HOME_BODY_0_RESULTS \
+  "    <div class=\"kiwixHomeBody\">\n" \
+  "        <style>\n" \
+  "          .book__list {\n" \
+  "            display: none;\n" \
+  "          }\n" \
+  "          .kiwixHomeBody {\n" \
+  "            justify-content: center;\n" \
+  "          }\n" \
+  "          .noResults {\n" \
+  "            font-size: 16px;\n" \
+  "            font-family: roboto;\n" \
+  "          }\n" \
+  "        </style>\n" \
+  "        <div class=\"noResults\">\n" \
+  "          No result. Would you like to <a href=\"?lang=\">reset filter</a>?\n" \
+  "        </div>\n" \
+  "        </style>\n" \
+  "        <div class=\"book__list\">\n" \
+  "        <h3 class=\"kiwixHomeBody__results\">0 book(s)</h3>\n" \
+  "        \n"
+
+#define RAY_CHARLES_DOWNLOAD \
+  "<!DOCTYPE html>\n" \
+  "<html lang=\"en\">\n" \
+  "<head>\n" \
+  "    <meta charset=\"UTF-8\">\n" \
+  "    <meta http-equiv=\"X-UA-Compatible\" content=\"IE=edge\">\n" \
+  "    <meta name=\"viewport\" content=\"width=device-width, initial-scale=1.0\">\n" \
+  "    <title></title>\n" \
+  "</head>\n" \
+  "<style>\n" \
+  "    .downloadLinksTitle {\n" \
+  "        text-align: center;\n" \
+  "        font-size: 32px;\n" \
+  "        margin-bottom: 8px;\n" \
+  "    }\n" \
+  "</style>\n" \
+  "<body>\n" \
+  "    <div class=\"downloadLinksTitle\">\n" \
+  "        Download links for <b><i>Ray Charles</i></b>\n" \
+  "    </div>\n" \
+  "    <a href=\"\" download>\n" \
+  "        <div>Direct</div>\n" \
+  "    </a>\n" \
+  "    <a href=\".sha256\" download>\n" \
+  "        <div>Sha256 hash</div>\n" \
+  "    </a>\n" \
+  "    <a href=\".magnet\" target=\"_blank\">\n" \
+  "        <div>Magnet link</div>\n" \
+  "    </a>\n" \
+  "    <a href=\".torrent\" download>\n" \
+  "        <div></div>\n" \
```

Comment:
The bug made its way into the unit-tests :)

Another thing that bothers me is the values of the `href` attributes in this HTML. Can we make the test more realistic? Probably it can be done easily by moving the noJS test into `test/library_server.cpp`.

Reply:
Ye done! Thanks